docs/madr/decisions/046-meshservice-policy-matching.md

```diff
@@ -0,0 +1,227 @@
+which will be covered in a separate MADR.
+
+Some things to remember:
+
+* Currently `kind: MeshService` in `from[].targetRef` is deprecated.
+* While `kind: MeshService` in `spec.targetRef` points to both a Dataplane and a set of inbounds,
+  it is not a canonical reference.
+  Its effects effectively mix with any policies applied
+  to any `MeshService` that points to this `Dataplane`/inbound.
+  A canonical reference can only be a direct choice of an entire `Dataplane` object
+  and potentially a subset of inbounds.
+* A `MeshService` only has a 1-1 relationship when used as `to[].targetRef.kind`
+  since then it corresponds to a particular Envoy outbound listener/some number of clusters.
+
+## Decision Drivers
+
+* Targeting headless `MeshService`
+* Targeting `MeshMultiZoneService`
+* A balance between correctness and ease of use when configuring policy.
+
+## Considered Options
+
+This MADR only consider references in `to[].targetRef.kind` and MeshHTTPRoute/MeshTCPRoute `backendRefs[].kind`.
+
+This MADR adds 1 new kind, `MeshMultiZoneService` and adds support
+for matching real `MeshService` resources to the existing `MeshService` kind.
+Matches on real `MeshService` objects have priority over `kuma.io/service`
+matches.
+
+A majority of the use cases for policy will involve selecting a specific *port* of the
+`MeshService`, `MeshServices` from a given Kuma zone and namespace.
+
+Note, we have already in targetRef:
+
+```yaml
+name: ...
+namespace: ...
+```
+
+Now we have to consider how we attach to a port and select from a zone.
+
+* Add top level field `zone` and `port` and `labels` for selecting groups of
+  `MeshService`
+* Add top level `port` field and use `labels: {"kuma.io/zone": "..."}`
+* Use generic Gateway API targetRef field `sectionName` to specify a port in
+  `targetRef` but `port` in `backendRef` and add `labels` for selecting
+  groups of `MeshService`.
+
+## Decision Outcome
+
+We choose:
+
+* use generic Gateway API targetRef field `sectionName` to specify `port` and
+  add `labels` but use `port` in `backendRefs`.
+
+### `targetRef`
+
+The policy is applied
+on both the Envoy outbound listeners and clusters
+that correspond to the `MeshService`.
+
+References for `kind: MeshService` have the following structure:
+
+```yaml
+---
+kind: MeshService
+spec:
+  ports:
+  - port: 80
+    name: http
+---
+targetRef:
+  kind: MeshService
+  namespace: backend
+  name: backend # only name or labels
+  labels: # only name or labels
+    kuma.io/zone: east
+  sectionName: http
+```
+
+The `name` of a `MeshService` ref always refers to the name of one, real Kubernetes
+object in the zone of the policy.
+
+Exactly one of either `targetRef.name` or `targetRef.labels` is **required**.
+
+#### `sectionName`
+
+If `sectionName` is set in `targetRef.kind` with `kind: MeshService`,
+it refers to an entry in `ports` by name and only traffic to that port is affected.
+
+In `backendRefs`, `sectionName` is required.
```

Comment:
you say here, that `sectionName` field is required in `backendRef`, but then in example you use `port` field. Which one is correct?

Reply:
ah yes, forgot to update this. it should be `port` in `backendRef`